src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -54,14 +54,27 @@ public void BeginRequest(HttpContext httpContext, HostingApplication.Context con
     {
         long startTimestamp = 0;
 
-        if (_eventSource.IsEnabled() || _metrics.IsEnabled())
+        if (_metrics.IsEnabled())
         {
-            context.EventLogOrMetricsEnabled = true;
+            context.MetricsEnabled = true;
             context.MetricsTagsFeature ??= new HttpMetricsTagsFeature();
```

Comment:
This gets allocated per request when metrics are on?

Reply:
It’s cached on the context